### PR TITLE
Add `WindowInfo.containerSize`

### PR DESCRIPTION
**Pull request by MatkovIvan**
## Proposed Changes

- Add `WindowInfo.containerSize` public API

It requires for max available space calculation in some material components.

## Testing

Test: Try to get window size via `LocalWindowInfo.current`

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (m-sasha) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt:51]: Add javadoc, and mention that the size is in pixels.

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (m-sasha) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: Shouldn't this be a "State"?

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (m-sasha) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: Why the change from `toInt` to `roundToInt`?

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: No

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: More correct one. Harmonization between platforms

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (m-sasha) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: Why not? The other properties are state. How can the user get notified about changes?

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (m-sasha) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: Why is it more correct? Seems that only uikit had `roundToInt`...

**Comment by m-sasha** [comment]
Consider adding a (skiko) unit test.

**Comment by m-sasha** [comment]
Maybe also add `WindowInfo.position`? See my request from 2022: https://github.com/JetBrains/compose-multiplatform/issues/1873

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: You probably right 🤔, I had an impression that `Local*.current` does this job

Comment (m-sasha) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: > You probably right 🤔, I had an impression that Local*.current does this job

It does, but only if you replace the whole (immutable) object when something changes, like (for example) in `LocalTextStyle`.

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt
Comment (igordmn) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt]: Small refactoring. We can write:

```
override var isWindowFocused: Boolean by mutableStateOf(false)

@ExperimentalComposeUiApi
override var size: IntSize by mutableStateOf(IntSize.Zero)

@ExperimentalComposeUiApi
override var keyboardModifiers: PointerKeyboardModifiers by GlobalKeyboardModifiers
```

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (igordmn) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: `roundToInt` in iOS/macOS is to prevent float rounding errors, because the size is float there.

`toInt` here is to not cut the content in case if we have non-integer density:
210*1.25 = 262.75

We shouldn't adapt the formula to other platforms, we should adapt it to how Swing internally converts the size of the window in pixels to int points.

It is impossible to revert int points to pixels precisely, so we have 2 evils - have 1px blank line (`toInt`), or sometime cut the content by 1px (`roundToInt`, sometimes it cuts, sometimes it adds a blank line). `toInt` is more stable here - it just always adds a blank line

Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: Reverted

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (igordmn) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]:
```suggestion
        val scale = component.density.density
```

There was a non-valid code here (that probably doesn't cause any bug yet). Because we use AWT property (component.width), we also should use AWT component's density, not scene's one (that can be different, if we expose configuration in the future).

**Review by igordmn — APPROVED** [review]
Let's wait for @m-sasha approval

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt
Comment (MatkovIvan) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt]: As you said, they are equal - it's called only after assignment component's density to scene. But OK, let's replace to `component` here

File: compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/awt/ComposeWindowTest.kt
Comment (m-sasha) [inline compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/awt/ComposeWindowTest.kt:160]: I think it's better to test `WindowInfo` separately from other things. A good test should validate one thing.

Also, don't you want to test it inside the composition (i.e. with `LocalWindow.current.containerSize`?

**Review by m-sasha — APPROVED** [review]

**Pull request merged**
